resources/steam.py

```diff
@@ -0,0 +1,85 @@
+import shlex
+import subprocess
+import requests
+
+from util import log
+
+
+def install(steam_exe_path, appid):
+    """
+    Calls Steam to install a game/app. This will display Steam's game install prompt, which displays install configurations and asks for confirmation.
+
+    :param steam_exe_path: path to the steam executable
+    :param appid: appid of the game/app to install
+    """
+    log('executing ' + steam_exe_path + ' steam://install/' + appid)
+    # https://developer.valvesoftware.com/wiki/Steam_browser_protocol
+    subprocess.call([steam_exe_path, 'steam://install/' + appid])
+
+
+def run(steam_exe_path, steam_launch_args, appid):
+    """
+    Calls Steam to run a game/app. This will run it, or in not installed display Steam install prompt
+
+    :param steam_exe_path: path to the steam executable
+    :param steam_launch_args: A string of Steam launch arguments (format "-arg1 -arg2 ...")
+    :param appid: appid of the game/app to run
+    """
+    user_args = shlex.split(steam_launch_args)
+    log('executing ' + steam_exe_path + ' ' + steam_launch_args + ' steam://rungameid/' + appid)
+
+    # https://developer.valvesoftware.com/wiki/Steam_browser_protocol
+    subprocess.call([steam_exe_path, user_args, 'steam://rungameid/' + appid])
```

Comment:
@AmineI sorry it took me so long to get around to testing this :bow: I had the following error in my logs when I tried to run a game:
```
2020-04-18 11:35:07.901 T:140570099152640   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.TypeError'>
                                            Error Contents: execv() arg 2 must contain only strings
                                            Traceback (most recent call last):
                                              File "/home/aaron/.kodi/addons/plugin.program.steam.library/addon.py", line 6, in <module>
                                                main.main()
                                              File "/home/aaron/.kodi/addons/plugin.program.steam.library/resources/main.py", line 210, in main
                                                plugin.run()
                                              File "/home/aaron/.kodi/addons/script.module.routing/lib/routing.py", line 130, in run
                                                self._dispatch(self.path)
                                              File "/home/aaron/.kodi/addons/script.module.routing/lib/routing.py", line 141, in _dispatch
                                                view_func(**kwargs)
                                              File "/home/aaron/.kodi/addons/plugin.program.steam.library/resources/main.py", line 114, in run
                                                steam.run(__addon__.getSetting('steam-exe'), __addon__.getSetting('steam-args'), appid)
                                              File "/home/aaron/.kodi/addons/plugin.program.steam.library/resources/steam.py", line 32, in run
                                                subprocess.call([steam_exe_path, user_args, 'steam://rungameid/' + appid])
                                              File "/nix/store/94kj1vifh3yyvm9vg4wwhghzmmz6dfr3-python-2.7.17/lib/python2.7/subprocess.py", line 172, in call
                                                return Popen(*popenargs, **kwargs).wait()
                                              File "/nix/store/94kj1vifh3yyvm9vg4wwhghzmmz6dfr3-python-2.7.17/lib/python2.7/subprocess.py", line 394, in __init__
                                                errread, errwrite)
                                              File "/nix/store/94kj1vifh3yyvm9vg4wwhghzmmz6dfr3-python-2.7.17/lib/python2.7/subprocess.py", line 1047, in _execute_child
                                                raise child_exception
                                            TypeError: execv() arg 2 must contain only strings
                                            -->End of Python script error report<--
```

Reply:
Don't be - we all have things to do, and this is definitely not the most important ;) ! 

About the issue, that's right, it slip past me, thanks for noticing it !
I can see why : It previously was 
` subprocess.call([__addon__.getSetting('steam-exe')] + userArgs + ['steam://rungameid/' + id])`
That concataned the arrays, effectively creating a big array with 
```[steam-exe, userArg1,userArg2, .... , steam://rungameid/id]```
What I did though, was create an array equivalent to 
```[steam-exe,[userArg1,userArg2,...],steam://rungameid/id]```

Problem being, there is now an array in an array. I'll fix this quickly by reverting to the previous behavior. Thanks again for noticing this !